includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -755,6 +761,41 @@ private function has_used_attributes( $attribute_names ) {
 		return true;
 	}
 
+	/**
+	 * Whether a given class is allowed to be styled in <amp-date-picker>.
+	 *
+	 * That component has child classes that won't be present in the document yet.
+	 * So get whether a class is an allowed child.
+	 *
+	 * @since 1.5.0
+	 * @link https://github.com/airbnb/react-dates/tree/05356/src/components
+	 *
+	 * @param string $class The name of the class to evaluate.
+	 * @return bool Whether the class is allowed as a child of <amp-date-picker>.
+	 */
+	private function is_class_allowed_in_amp_date_picker( $class ) {
+		// Some prefixes, some full class names.
+		$allowed_class_beginnings = [
+			'CalendarDay',
+			'CalendarMonth',
+			'DateInput',
+			'DateRangePicker',
+			'DayPicker',
```

Comment:
These classes could be more specific, though that would mean a longer array.

For example, instead of just `DayPicker`, this array could have `DayPicker_`, `DayPickerKeyboardShortcuts_`, `DayPickerNavigation_`, etc...

This will of course still accept those, though it'll also accept something like `DayPickerRandomIncorrect`.

Reply:
They're now all specific with https://github.com/ampproject/amp-wp/pull/4339/commits/5bffd65ecba76846919095ea883c93d6df20954a, like `DayPickerNavigation` instead of only `DayPicker`